### PR TITLE
Amazons

### PR DESCRIPTION
**Pull request by Noordfrees**

### CODE REVIEW

**Comment by gunchleoc** [comment]
I am doing the review offline and will simply push my suggestions to this branch.

I have done the units.lua files and fixed the following inconsistencies - listing them here for me for when I start working on the individual init.lua files:

- cassavaroot => cassava root
- amazons => Amazons
- warriors gathering => warriors’ gathering
- stonecarver => stone carver
- Stonebowl => Stone bowl
- hunter gatherer => hunter-gatherer

Also "carvery" sounds like a place where you get food: https://en.wikipedia.org/wiki/Carvery
Shall we use "Stonemasonry" instead of "Stonecarvery"? Or "Stone Carver's Hut"?

File: data/tribes/initialization/amazons/init.lua [hunk 1]
```diff
@@ -11,7 +11,7 @@ local r = {
    author = _"The Widelands Development Team",
    -- TRANSLATORS: This is a tribe name
    descname = _"Amazons (experimental)",
-   tooltip = _"The people of the amazonian jungle. Experimental state yet",
+   tooltip = _"The people of the amazonian jungle. Still in experimental state.",
```
Comment (hessenfarmer) [inline data/tribes/initialization/amazons/init.lua]: Well we could pretend we left experimental and went to under development like with the frisians

File: data/tribes/initialization/atlanteans/units.lua [hunk 2]
```diff
@@ -1234,91 +1234,91 @@ descriptions:new_tribe {
       {
          name = "deadtree7",
          helptexts = {
-            -- TRANSLATORS: Helptext for an atlantean immovable: Ashes
+            -- TRANSLATORS: Helptext for an atlantean immovable: Dead Tree
```
Comment (hessenfarmer) [inline data/tribes/initialization/atlanteans/units.lua]: if we leave atalanteans here we should change it for the other tribes. Alternative we should describe that htis is an amazonian immovable used by the other tribes

File: data/tribes/initialization/amazons/units.lua [hunk 3]
```diff
@@ -1282,17 +1282,17 @@ descriptions:new_tribe {
...
          }
       },
       {
          name = "amazons_hunter_gatherers_hut",
          helptexts = {
-            -- TRANSLATORS: Purpose helptext for an amazon production site: Hunter Gatherer's Hut
-            purpose = pgettext("building", "Hunts animals to produce meat. Catches fish in the waters."),
-            -- TRANSLATORS: Note helptext for an amazon production site: Hunter Gatherer's Hut
+            -- TRANSLATORS: Purpose helptext for an amazon production site: Hunter-Gatherer's Hut
+            purpose = pgettext("builamazons_buildingding", "Hunts animals to produce meat. Catches fish in the waters."),
```
Comment (hessenfarmer) [inline data/tribes/initialization/amazons/units.lua]: typo

File: data/tribes/initialization/amazons/units.lua [hunk 4]
```diff
@@ -1408,7 +1408,7 @@ descriptions:new_tribe {
          name = "amazons_food_preserver",
          helptexts = {
             -- TRANSLATORS: Purpose helptext for an amazon production site: Food Preserver
-            purpose = pgettext("building", "Prepares rations to feed the scouts and miners."),
+            purpose = pgettext("buiamazons_buildinglding", "Prepares rations to feed the scouts and miners."),
```
Comment (hessenfarmer) [inline data/tribes/initialization/amazons/units.lua]: typo

**Review by hessenfarmer — COMMENTED** [review]
some small nits I will take care of them myself

**Comment by hessenfarmer** [comment]
> Also "carvery" sounds like a place where you get food: https://en.wikipedia.org/wiki/Carvery
> Shall we use "Stonemasonry" instead of "Stonecarvery"? Or "Stone Carver's Hut"?

I would name this Stone Workshop as we are producing tools and weapons instead of building material like a mason does

**Comment by gunchleoc** [comment]
Instead of having multiple copies of the same tree just for the terrain affinity, how about removing the terrain affinity and controlling it my removal chance, like for the Cocoa tree? This would unclutter the immovable help.

How about merging ironwood with blackwood? Blackwood used to be called ironwood once, so we could return to that for the Barbarians.

**Comment by hessenfarmer** [comment]
> 
> 
> Instead of having multiple copies of the same tree just for the terrain affinity, how about removing the terrain affinity and controlling it my removal chance, like for the Cocoa tree? This would unclutter the immovable help. 

Well we would loose terrain affinity then. Resulting in being able to grow the trees on any ground with equal chance which is not intended as they are still trees (i.e. they have the tree attribute). Cocoa tree is more like a crops or like a bush from a game mechanics point of view. So I think we should think about a solution to have both Terrain affinity and uncluttered help

**Comment by gunchleoc** [comment]
OK, let's keep the trees for now and leave uncluttering the help to another day.

* How about merging ironwood with blackwood? Blackwood used to be called ironwood once, so we could return to that for the Barbarians.
* Having charcoal rather than coal means one less ware to trade when we finally finish the trading implementation. It is also a bit confusing because all other tribes' charcoal is simply coal.
* Rope Weaver Booth => Rope Weaver's Booth
* Rare Tree Woodcutter's Hut needs name uncluttering. => Rare Tree Cutter's Hut / Rare Woodcutter's Hut / Precious Woodcutter's Hut?
* stonetipped => stone-tipped

**Comment by hessenfarmer** [comment]
> 
> 
> OK, let's keep the trees for now and leave uncluttering the help to another day.
> 
>     * How about merging ironwood with blackwood? Blackwood used to be called ironwood once, so we could return to that for the Barbarians.

Technically ironwood is harvested while blacckwood is refined. I wouldn't object on this change though but this should be discussed in the forum to have King_of_Nowhere included, who is the spiritus rector of this tribe.
> 
>     * Having charcoal rather than coal means one less ware to trade when we finally finish the trading implementation. It is also a bit confusing because all other tribes' charcoal is simply coal.

I can't remeber exactly, but I believe this was made due to restrictions in the old help. I will have a look in the commits whetehr I left a comment on this. Should be discussed in the forum as well.  
Edit: Couldn't find any reason to not make them the same anymore. Still needs consensus in forums I think

> 
>     * Rope Weaver Booth => Rope Weaver's Booth
> 
>     * Rare Tree Woodcutter's Hut needs name uncluttering. => Rare Tree Cutter's Hut / Rare Woodcutter's Hut / Precious Woodcutter's Hut?

I am open for any suggestion as well as for other complicated names, but again should be fixed in the forum.

> 
>     * stonetipped => stone-tipped

**Comment by matthiakl** [comment]
While playing a game with amazons, I noticed some bugs. Not sure whether you want to address them now or after the merge.

- Treetop sentries cannot be placed on every tree. The problem was already mentioned in the forum (https://www.widelands.org/forum/topic/2897/?page=37#post-34333)
- Removing a treetop sentry constructionsite (with no materials) burns down the tree. Removing a normal constructionsite however leaves a clean field behind, so I think the tree should be preserved.
- Geologists send messages for iron and coal, even though the spots on the map are marked with "No resources"

**Comment by Noordfrees** [comment]
I would rather address them after the merge. The diff is too large to allow reviewing non-trivial coding changes reasonably, and these issues are not critical. Open an issue please

**Comment by hessenfarmer** [comment]
I just implemented the agreed changes from the forum.
If @gunchleoc is fine with this and she finished proofreading this should go in from my side.

**Review by gunchleoc — APPROVED** [review]
Consistency check done & crashes fixed.

The seafaring starting conditions don't work yet, but we can do those in a follow-up branch.

Take a look at my changes and merge if you're happy with them.

**Comment by hessenfarmer** [comment]
Happy with your changes. However I recognised some Copy paste issues in the translator helptexts that I created in the very beginning. I need to search for the atlantean keyword over all files again to be sure we change them all correctly.   
I'll do this tonight.
Seafaring sc are not implemented yet. But I am planning to do a Startcondition fixing branch for all tribes based on some excel calculations made after this got in, where I can cover this.

**Pull request merged**